gui/src/views/ViewBackups.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+    <div>
+        <Backups :backups="backups" :backups_by_snapshot="backups_by_snapshot" :cluster="cluster" @emitRestoreParams="setRestoreParams($event)" />
+        <CreateBackup :cluster="cluster" :backups="backups" :snapshot_ids="snapshot_ids" @updateBackupList="fetchData"/>
+        <RestoreCluster
+                :cluster="cluster"
+                :snapshot_id="snapshot_id"
+                :backup_type="backup_type"
+                :changeRestoreType=changeRestoreType
+                @createRestoreRequestBody="createRestoreRequestBody($event)" />
+        <ConfirmRestore :cluster_id="cluster_id" :snapshot_id="snapshot_id" :data="restore_request_body"/>
+    </div>
+</template>
+
+<script>
+    import Backups from '../components/Backups'
+    import CreateBackup from '../components/CreateBackup'
+    import RestoreCluster from '../components/RestoreCluster';
+    import ConfirmRestore from '../components/ConfirmRestore';
+
+    export default {
+      name: 'ViewBackups',
+      components: {
+        RestoreCluster,
+        Backups,
+        CreateBackup,
+        ConfirmRestore
+      },
+      data() {
+        return {
+          cluster_id: this.$route.params.cluster_id,
+          cluster: {},
+          backups: {},
+          backups_by_snapshot: [[]],
+          snapshot_ids: Set,
+          snapshot_id: '',
+          backup_type: 0,
+          restore_request_body: {},
+          autoRefreshTimer: ''
+        };
+      },
+      mounted() {
+        this.fetchData();
+        this.autoRefreshTimer = setInterval(this.fetchData, 10000);
+
+      },
+      beforeDestroy() {
+        clearInterval(this.autoRefreshTimer);
+      },
+      methods: {
+        changeRestoreType(type) {
+          this.restore_type = type;
+        },
+        fetchData() {
+          this.$api.get(`clusters/${this.cluster_id}`)
+          .then((response) => {
+            if (response.status === 200) {
+              this.cluster = response.data.entries[0];
+            }
+          });
+          this.$api.get(`clusters/${this.cluster_id}/backups`)
+          .then((response) => {
+            if (response.status === 200) {
```

Comment:
If it doesn't return `200`, what will happen ?

This if block can be less nested by doing something like this ?
```
 if (response.status !== 200) {
  // do something ? and return 
}

// do the same things but less nested
this.backups = response.data;

Reply:
In this case I guess nothing will happen if the request doesn't succeed. I can add some kind of console log or add some feedback on the screen indicating the gui failed to get information from cassy.

Reply:
Actually, I think if we want to make it do something when the response isn't `200`, it would be a `catch` block like I mentioned in my other comment.